### ISSUE TITLE
Missing else part on constexpr variant(variant const& rhs); effects

### ISSUE THREAD

**Issue by viboes** [eggs-cpp/variant#5]
In
```
constexpr variant(variant const& rhs);

    Requires: std::is_copy_constructible_v<T> is true for all T in Ts....
    Effects: If rhs has an active member of type T, initializes the active member as if direct-non-list-initializing an object of type T with the expression *rhs.target<T>().
```

you need to add how the constructor behaves is `rhs`has no active member.

**Comment by viboes** [comment]
The same applies to other constructors.

**Comment by K-ballo** [comment]
Good catch. I derived the wording from Optional objects for [LFTS](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4480.html#optional). I will report it back to them and follow their resolution.

**Comment by K-ballo** [comment]
Consulted with LWG members, the postconditions mandate the effects on empty.

However, by now I don't like it either, I find it too weak. I will look into changing it (`eggs::variant`, not `std::experimental::optional`).

**Comment by viboes** [comment]
Oh, I see. I didn't know that a way to state the effects was via post-conditions.
We learn something new every day :) 
Please be free to close this issue.

**Comment by K-ballo** [comment]
Fixed by 85c0e4635262d07cd1161aa51c994973b895e4e7
